pulp_smash/pulp3/pytest_plugin/__init__.py

```diff
@@ -252,6 +266,57 @@ def tasks_api_client(pulpcore_client):
     return TasksApi(pulpcore_client)
 
 
+@pytest.fixture
+def status_api_client(pulpcore_client):
+    return StatusApi(pulpcore_client)
+
+
+@pytest.fixture
+def stop_and_check_services(status_api_client):
+    """Stop services and wait up to 10 seconds to check if services have stopped."""
+
+    def _stop_and_check_services(pulp_services=None):
+        svc_mgr.stop(pulp_services or PULP_SERVICES)
```

Comment:
I'm wondering, if the waiting should be done in the `smash_cli.ServiceManager`.

Reply:
I agree, it should probably be implemented as `def stop(services, blocking=True)`

Reply:
I'm not sure about that. 
`smash_cli.ServiceManager` is generic enough to stop and start any service, e.g. posgresql, and not necessarily pulp.
Here all the logic for checking and waiting is specifically for pulp.
I think either we keep it as is in fixtures, or we need to provide some callback to service manager for waiting and checking according to the services it is working with. I'd vote for keeping as is.

Thougths? @mdellweg @dralley @dkliban

Reply:
I would leave it as is.

Reply:
That's a good point. Keep it this way then.